src/main/java/DAO/UserDAO.java

```diff
@@ -1,206 +1,78 @@
-            System.out.println(e);
-            throw new RuntimeException(e);
-
-        }
-        finally {
-            try {
-                preparedStatement.close();
-                connection.close();
-            } catch (SQLException e) {
-                throw new RuntimeException(e);
-            }
-        }
+    public List<User> getAll() {
+        return jdbcTemplate.query(FETCH_ALL_USERS_SQL, new UserMapper());
     }
 
     @Override
     public void save(User user) {
-        try {
-            connection = dataSource.getConnection();
-            preparedStatement = connection.prepareStatement(SAVE_USER_SQL);
-            preparedStatement.setString(1, user.getID().toString());
-            preparedStatement.setString(2, user.getName());
-            preparedStatement.setTimestamp(3, user.getCreationDate());
-            preparedStatement.setString(4, user.getRole().name());
-            preparedStatement.executeUpdate();
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
-        finally {
-            try {
-                preparedStatement.close();
-                connection.close();
-            } catch (SQLException e) {
-                throw new RuntimeException(e);
-            }
-        }
+        jdbcTemplate.update(SAVE_USER_SQL, user.getId(), user.getName(), user.getCreationDate(), user.getRole().name());
+        System.out.println(user.getRole().name());
     }
 
-    @Override
-    public Optional<User> fetchByID(UUID id)  {
-        try {
-            connection = dataSource.getConnection();
-            preparedStatement = connection.prepareStatement(FETCH_USER_BY_ID_SQL);
-            preparedStatement.setString(1, id.toString());
-            resultSet = preparedStatement.executeQuery();
-
-            while (resultSet.next()) {
-                UUID columnID = UUID.fromString(resultSet.getString("id"));
-                String columnName = resultSet.getString("name");
-                Timestamp columnCreationDate = resultSet.getTimestamp("creation_date");
-
-                switch (resultSet.getString("user_role")) {
-                    case "CLIENT" :
-                        return Optional.of(new Client(columnID, columnName, columnCreationDate));
-                    case "ADMIN" :
-                        return Optional.of(new Admin(columnID, columnName, columnCreationDate));
-                }
-
-            }
-        } catch (SQLException e) {
-            throw new RuntimeException(e);
-        }
-        finally {
-            try {
-                resultSet.close();
-                preparedStatement.close();
-                connection.close();
-            } catch (SQLException e) {
-                throw new RuntimeException(e);
-            }
-        }
-        return Optional.empty();
+    public void deleteAll() {
+        jdbcTemplate.execute(DELETE_ALL_USERS_SQL);
+    }
+
+    public void updateID(UUID userID, UUID newID) {
```

Comment:
not clear name of method. better use smth like `updateUserIdAndCreateTicket`

Reply:
Fixed